onebusaway-android/src/main/java/org/onebusaway/android/ui/TripPlanFragment.java

```diff
@@ -0,0 +1,464 @@
+
+
+        if (myCalendar == null) {
+            Date date = builder.getDateTime();
+            if (date == null)
+                date = new Date();
+            myCalendar = Calendar.getInstance();
+            myCalendar.setTime(date);
+        }
+
+
+        resetDateTimeLabels();
+
+        boolean leaving = builder.getArriveBy();
+        String leavingString = getString(leaving ? R.string.trip_plan_arriving : R.string.trip_plan_leaving);
+        leavingChoice.setText(leavingString);
+    }
+
+    @Override
+    public void onCreateOptionsMenu(Menu menu, MenuInflater inflater) {
+        super.onCreateOptionsMenu(menu, inflater);
+        inflater.inflate(R.menu.menu_trip_plan, menu);
+    }
+
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        int id = item.getItemId();
+
+        if (id == R.id.action_settings) {
+            advancedSettings();
+            return true;
+        }
+
+        return super.onOptionsItemSelected(item);
+    }
+
+
+    private void advancedSettings() {
+
+        AlertDialog.Builder dialogBuilder = new AlertDialog.Builder(getActivity())
+                .setView(R.layout.trip_plan_advanced_settings_dialog);
+
+        final AlertDialog dialog = dialogBuilder.create();
+
+        dialog.show();
+
+        final TypedArray transitModeResource = getContext().getResources().obtainTypedArray(R.array.transit_mode_array);
+        final CheckBox minimizeTransfersCheckbox = (CheckBox) dialog.findViewById(R.id.checkbox_minimize_transfers);
+        final Spinner spinnerTravelBy = (Spinner) dialog.findViewById(R.id.spinner_travel_by);
+        final CheckBox wheelchairCheckbox = (CheckBox) dialog.findViewById(R.id.checkbox_wheelchair_acccesible);
+        final EditText maxWalkEditText = (EditText) dialog.findViewById(
+                R.id.number_maximum_walk_distance);
+
+        minimizeTransfersCheckbox.setChecked(builder.getOptimizeTransfers());
+
+        wheelchairCheckbox.setChecked(builder.getWheelchairAccessible());
+
+        ArrayAdapter adapter = ArrayAdapter.createFromResource(getActivity(),
+                R.array.transit_mode_array, R.layout.advanced_settings_spinner_item);
+        adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+        spinnerTravelBy.setAdapter(adapter);
+
+        int modeSetId = builder.getModeSetId();
+        if (modeSetId != -1) {
+            for (int i = 0; i < transitModeResource.length(); i++) {
+                if (transitModeResource.getResourceId(i, -1) == modeSetId) {
+                    spinnerTravelBy.setSelection(i);
+                    break;
+                }
+            }
+        }
+
+        final boolean unitsAreImperial = !RegionUtils.getUnitsAreMetric(getContext());
+
+        Double maxWalk = builder.getMaxWalkDistance();
+        if (maxWalk != null && Double.MAX_VALUE != maxWalk) {
+            if (unitsAreImperial)
+                maxWalk *= 3.281;
```

Comment:
Can you please refactor this magic number out to a `final static` variable with a name for what is it?  Even better would be moving the conversion to it's own utility method in the `ConversionUtils` class, with a method signature like:

`public static double metersToFeet(double meters) {`